From 646eab9e2b04683d1eee550eb535a10250ca84f3 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Thu, 27 Jun 2019 23:14:42 +0800
Subject: [PATCH 4/4] platform/x86: sony-laptop: restore keyboard backlight
 status on resume

Some VAIO laptops reset the keyboard backlight status when suspend/resume.

Restore the keyboard backlight status when resuming.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 drivers/platform/x86/sony-laptop.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/drivers/platform/x86/sony-laptop.c b/drivers/platform/x86/sony-laptop.c
index c96211182c05..3e461e86cb48 100644
--- a/drivers/platform/x86/sony-laptop.c
+++ b/drivers/platform/x86/sony-laptop.c
@@ -130,6 +130,7 @@ static int sony_nc_kbd_backlight_setup(struct platform_device *pd,
 		unsigned int handle);
 static void sony_nc_kbd_backlight_cleanup(struct platform_device *pd,
 		unsigned int handle);
+static void sony_nc_kbd_backlight_resume(void);
 
 static int sony_nc_battery_care_setup(struct platform_device *pd,
 		unsigned int handle);
@@ -1540,6 +1541,14 @@ static void sony_nc_function_resume(void)
 		case 0x0135:
 			sony_nc_rfkill_update();
 			break;
+		case 0x0137:
+		case 0x0143:
+		case 0x014b:
+		case 0x014c:
+		case 0x0153:
+		case 0x0163:
+			sony_nc_kbd_backlight_resume();
+			break;
 		default:
 			continue;
 		}
@@ -1972,6 +1981,18 @@ static void sony_nc_kbd_backlight_cleanup(struct platform_device *pd,
 	}
 }
 
+static void sony_nc_kbd_backlight_resume(void)
+{
+	if (!kbdbl_ctl)
+		return;
+
+	__sony_nc_kbd_backlight_mode_set(kbdbl_ctl->mode);
+
+	if (kbdbl_ctl->has_timeout) {
+		__sony_nc_kbd_backlight_timeout_set(kbdbl_ctl->timeout);
+	}
+}
+
 struct battery_care_control {
 	struct device_attribute attrs[2];
 	unsigned int handle;
-- 
2.21.0

